File: Challenge#03/Challenge_03.py
# ** Tu desafío: **
# Determina cuántas claves de cifrado son válidas según sus políticas.


cont_failures = 0

with open("Challenge#03/message_03.txt", "r") as file:
    f = list(file)
    for line in f:

        letter_cont = 0
        cipher, password = line.split(":")
        password = (password.split("\n"))[0]
        password = password.strip()
        numbers, letters = cipher.split(" ")
        minimum, maximum = numbers.split("-")
        minimum = int(minimum)
        maximum = int(maximum)

        for letter in password:
            if letter == letters:
                letter_cont += 1

        if letter_cont < minimum or letter_cont > maximum:
            cont_failures += 1
            if cont_failures == 42:
                print(password)

            # Pista para secreto
            if cont_failures == 13:
                print(password)

File: Challenge#03/Challenge_03.js
const fs = require('fs')

const input = fs.readFileSync('Challenge#03/message_03.txt', 'utf8').split('\n')

// Contador de contraseñas invalidas 
let error_count = 0

for (const line of input) {
    // Separación de variables
    const [key, password] = line.split(':')
    const [numbers, letter] = key.split(" ")
    const [minimum, maximum] = numbers.split("-")

    // Contar ocurrencias de la letra en la contraseña
    let letter_count = 0
    for (const char of password) {
        if (char === letter) {
            letter_count++
        }
    }

    // Verificar si la cantidad de ocurrencias está dentro del rango
    if (letter_count < minimum || letter_count > maximum) {
        error_count++
    }

    // Imprimir contraseñas en caso de error_count igual a 42 o 13
    if (error_count === 42 || error_count === 13) {
        console.log(password)
    }
}
